src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -73,13 +81,30 @@ public GoogleBucketResource getOrCreateBucketForFile(Dataset dataset,
             billingProfile,
             null);
 
-        return bucketService.getOrCreateBucket(
-            dataLocationSelector.bucketForFile(datasetName, billingProfile),
-            projectResource,
+        List<UUID> bucketsForDataset = datasetBucketDao.getBucketForDatasetId(UUID.fromString(datasetId));
```

Comment:
The caller has a `UUID` but needs to convert it to a `String`, and then we convert it back to a `UUID` here. I think it would be better to have the method accept `UUID` instead.